@types/nsfw/index.d.ts

```diff
@@ -0,0 +1,50 @@
+declare function nsfw(
+  dir: string,
+  callback: nsfw.EventsCallback,
+  options?: nsfw.WatcherOptions
+): Promise<nsfw.Watcher>;
+
+declare namespace nsfw {
+  export class Watcher {
+    start(): Promise<void>;
+    stop(): Promise<void>;
+  }
+  export interface BaseEvent {
+    action: number;
+    directory: string;
+  };
+  export interface CreatedEvent extends BaseEvent {
+    action: nsfw.actions.CREATED;
+    file: string;
+  }
+  export interface DeletedEvent extends BaseEvent {
+    action: nsfw.actions.DELETED;
+    file: string;
+  }
+  export interface ModifiedEvent extends BaseEvent {
+    action: nsfw.actions.MODIFIED;
+    file: string;
+  }
+  export interface RenamedEvent extends BaseEvent {
+    action: nsfw.actions.RENAMED;
+    oldFile: string;
+    newDirectory: string;
+    newFile: string;
+  }
+  export type Event = CreatedEvent | DeletedEvent | ModifiedEvent | RenamedEvent;
+  export type EventsCallback = (events: Event[]) => void;
+  export interface WatcherOptions {
+    debouceMS?: number;
+    errorCallback?(errors: Error[]);
+  };
+  export enum actions {
```

Comment:
Is this your own naming or does the nsfw user lowercase?  If you named this, then it’s probably best to use uppercase Actions instead.

Reply:
Event better, use the singular “Action” instead

Reply:
`nsfw` does indeed name it like this, so changing it would make `console.log(nsfw.actions)` not work correctly I think?